Source/Core/Core/AchievementManager.cpp

```diff
@@ -212,6 +279,70 @@ AchievementManager::ResponseType AchievementManager::FetchGameData()
       rc_api_process_fetch_game_data_response);
 }
 
+AchievementManager::ResponseType AchievementManager::FetchUnlockData(bool hardcore)
+{
+  rc_api_fetch_user_unlocks_response_t unlock_data{};
+  std::string username = Config::Get(Config::RA_USERNAME);
+  std::string api_token = Config::Get(Config::RA_API_TOKEN);
+  rc_api_fetch_user_unlocks_request_t fetch_unlocks_request = {.username = username.c_str(),
+                                                               .api_token = api_token.c_str(),
+                                                               .game_id = m_game_id,
+                                                               .hardcore = hardcore};
+  ResponseType r_type =
+      Request<rc_api_fetch_user_unlocks_request_t, rc_api_fetch_user_unlocks_response_t>(
+          fetch_unlocks_request, &unlock_data, rc_api_init_fetch_user_unlocks_request,
+          rc_api_process_fetch_user_unlocks_response);
+  if (r_type == ResponseType::SUCCESS)
+  {
+    m_lock.lock();
```

Comment:
Seems like a normal:

```cpp
std::lock_guard lg{m_lock};
```

declaration here would allow you to get rid of the explicit lock and unlock